conjure-core/src/main/java/com/palantir/conjure/defs/ConjureTypeParserVisitor.java

```diff
@@ -72,7 +85,13 @@ public Type resolve(LocalReferenceType reference) {
         @Override
         public Type resolve(ForeignReferenceType reference) {
             String namespaceFile = importProviders.get(reference.namespace());
-            Preconditions.checkNotNull(namespaceFile, "Import not found for namespace: %s", reference.namespace());
+            Preconditions.checkNotNull(
+                    namespaceFile,
+                    "Import not found for namespace: %s%s",
+                    reference.namespace(),
+                    conjureSourceFilePath
+                            .map(file -> String.format(", encountered while processing transitive file: '%s'", file))
```

Comment:
I think something like "imported" is more descriptive. 

```suggestion
                            .map(file -> String.format(", encountered while processing imported file: '%s'", file))
```